src/Tools/dotnet-monitor/LibrarySharing/DefaultSharedLibraryInitializer.cs

```diff
@@ -86,6 +152,22 @@ public IFileProviderFactory Initialize()
             return new Factory(sharedLibraryPath);
         }
 
+        private static int ComputeHashAndLength(SafeFileHandle handle, Span<byte> destination)
+        {
+            int length = (int)RandomAccess.GetLength(handle);
```

Comment:
How does that benefit this though? Also, the current usage allows me to use a stack allocated array for the hash values whereas those chunk reading APIs return a new array for each computation.

Reply:
I'm think that instead of hashing both the source and target file (which requires reading of the entire content and computing a hash value for each) that I change this to doing some type of chunked simultaneous reading of the content of the source and target files and comparing the chunks; if any chunk of data is different, then immediately fail. I think this should be faster in general and fails faster in the failure case.